mumax3c/scripts/energy.py

```diff
@@ -8,13 +9,30 @@
 def energy_script(system, ovf_format):
     mx3 = ""
     for term in system.energy:
-        if len(system.energy.get(type=type(term))) > 1:
+        if isinstance(term, mm.Zeeman):  # Handled separately
+            continue
+        elif len(system.energy.get(type=type(term))) > 1:
             raise RuntimeError(
-                "Mumax3 does not allow more than one energy term of the same class."
+                "Mumax3 does not allow more than one energy term of the same class "
+                "except for the Zeeman term."
             )
-        mx3 += globals()[f"{term.__class__.__name__.lower()}_script"](
-            term, system, ovf_format
-        )
+        else:
+            mx3 += globals()[f"{term.__class__.__name__.lower()}_script"](
+                term, system, ovf_format
+            )
+
+    num_zeeman_vectors = 0
+    if zeeman_terms := system.energy.get(type=mm.Zeeman):
+        for term in zeeman_terms:
+            if isinstance(term.H, (tuple, list, dict, np.ndarray)):
+                num_zeeman_vectors += 1
+                if num_zeeman_vectors > 1:
+                    H_field = df.Field(mesh=system.m.mesh, dim=3, value=term.H)
+                    mx3 += zeeman_script(mm.Zeeman(H=H_field), system, ovf_format)
+                else:
+                    mx3 += zeeman_script(term, system, ovf_format)
```

Comment:
We just silently ignore Zeeman terms with the wrong `H` type. Maybe we should raise an error if we detect a wrong type?

Reply:
Sorry, I did not understand the question. What exactly do you mean by wrong `H` type? The type of `H` is checked in `micromagneticmodel` by `ubermagutil.typesystem` as `H=ts.Parameter(descriptor=ts.Vector(size=3), otherwise=df.Field)`. Does this answer the question?

Reply:
Yes, it does. Different question: What happens if `H` is of type `df.Field`?

Reply:
If `H` is `df.Field`, we write an `ovf` file representing the field which is read by Mumax3 (by `Bext.Add()` to be exact). The relevent code is in `util.py`.